main.py

```diff
@@ -0,0 +1,159 @@
+            image, image_tensor = load_image(image_uri)
+            # run grounding dino model
+            logging.info(f"Running GroundedDINO with DETECTION PROMPT {prompt}.")
+            boxes_filt, pred_phrases = get_grounding_output(
+                model, image_tensor, prompt, 0.3, 0.25, device=args.device
+            )
+            # denormalize boxes (from [0, 1] to image size)
+            H, W, _ = image.shape
+            for i in range(boxes_filt.size(0)):
+                boxes_filt[i] = boxes_filt[i] * torch.Tensor([W, H, W, H])
+                boxes_filt[i][:2] -= boxes_filt[i][2:] / 2
+                boxes_filt[i][2:] += boxes_filt[i][:2]
+
+            predictor.set_image(image)
+            run_segmentation(predictor, image, boxes_filt, prompt)
+
+
+def log_video_segmentation(args, model: GroundingDINO, predictor: Sam):
+    video_path = args.video_path
+    assert video_path.exists()
+    cap = cv2.VideoCapture(str(video_path))
+
+    prompt = args.prompts
+
+    idx = 0
+    while cap.isOpened():
+        ret, bgr = cap.read()
+        if not ret:
+            break
+        rr.set_time_sequence("frame", idx)
+        rgb = cv2.cvtColor(bgr, cv2.COLOR_BGR2RGB)
+        rgb = resize_img(rgb, 512)
+
+        for prompt in args.prompts:
+            image_tensor = image_to_tensor(rgb)
+            # run grounding dino model
+            logging.info(f"Running GroundedDINO with DETECTION PROMPT {prompt}.")
+            boxes_filt, pred_phrases = get_grounding_output(
+                model, image_tensor, prompt, 0.3, 0.25, device=args.device
+            )
+            # denormalize boxes (from [0, 1] to image size)
+            H, W, _ = rgb.shape
+            for i in range(boxes_filt.size(0)):
+                boxes_filt[i] = boxes_filt[i] * torch.Tensor([W, H, W, H])
+                boxes_filt[i][:2] -= boxes_filt[i][2:] / 2
+                boxes_filt[i][2:] += boxes_filt[i][:2]
+
+            predictor.set_image(rgb)
+            run_segmentation(predictor, rgb, boxes_filt, prompt)
+
+        idx += 1
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Run IDEA Research Grounded Dino + SAM example.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--model",
+        action="store",
+        default="vit_b",
+        choices=MODEL_URLS.keys(),
+        help="Which model to use."
+        "(See: https://github.com/facebookresearch/segment-anything#model-checkpoints)",
+    )
+    parser.add_argument(
+        "--device",
+        action="store",
+        default="cpu",
+        help="Which torch device to use, e.g. cpu or cuda. "
+        "(See: https://pytorch.org/docs/stable/tensor_attributes.html#torch.device)",
+    )
+
+    parser.add_argument(
+        "--prompts",
+        default=["tires", "windows"],
+        nargs="+",
+        type=str,
+        help="List of prompts to use for bounding box detection.",
+    )
```

Comment:
You can use multi-word prompts. It would look like this `python main.py --prompts "tires and floors" "windows"` so you just need to separate out by " but it can be as long as you'd like